src/server/req_delete.c

```diff
@@ -1564,24 +1564,22 @@ post_delete_mom1(struct work_task *pwt)
 				return; /* will be back when replies */
 			goto resend;
 		} else if (rc == PBSE_UNKJOBID) {
-			/* if job was in prerun state, cannot delete it, even if mom does not know
-			 * about this job. Going ahead and deleting could result in a
-			 * server crash, when post_sendmom completes.
-			 */
+
 			if (check_job_substate(pjob, JOB_SUBSTATE_PRERUN)) {
-				if (pjob->ji_pmt_preq != NULL)
-					reply_preempt_jobs_request(rc, PREEMPT_METHOD_DELETE, pjob);
-				if (update_deljob_rply(preq_clt, pjob->ji_qs.ji_jobid, rc))
-					req_reject(rc, 0, preq_clt);
-				return;
+				/* This means the job has trigerred to run again,
+				 try deleting again! */
+				rc = issue_signal(pjob, sigtj, post_delete_mom1, preq_clt);
```

Comment:
Looking at the previous comment, it seems safer to just error out the deletion of this job and leaving it as it was. Did you test to make sure that the server doesn't crash in this scenario? Also, use the 'tries' counter to only do this once more, just like the if (rc == PBSE_UNKSIG) case to avoid getting stuck in a loop of mom rejecting it and server issuing it again.

Reply:
I did test it out to make sure that it doesn't crash.
The counter was used for unknown signals. In this case, we are trying to delete a job that was tried to run again and on the way to mom. It could result in it more than one attempt if the mom ends up rejecting the job and the scheduler did run it again on the same mom before the mom responds to the delete signal. But run_count should prevent that from going on forever.